### PR TITLE
Api gateway services

### PR DESCRIPTION
**Pull request by faad555**
created three Api-services 
I followed MVC structure. each have it's controller where we can add deferent actions later example ../profile/view, ../profile/create etc
and each service have its own model that have its methods.

### CODE REVIEW

**Review by Silver-Golden — CHANGES_REQUESTED** [review]
Overall grand, however there are two main things

1. You are using eclipse but didnt update teh ``.gitignore`` to reflect this, as such a mountain of (local) config files and compiled binaries got added
    ```gitgnore
    .classpath
    .project
    /.settings
    /**/.settings
    /target
    /**/target
    ```
2. The microservices are in teh wrong locaiton, they should be inside the ``Microservices`` folder

Fix those issues then come back for a review of the code without it being cluttered up by all the excess files

**Comment by M-Byte480** [comment]
I am little confused on what the proposed code provides?

There is not routing or distribution logic it appears. Quick google search gave me this article: https://medium.com/@ankithahjpgowda/api-gateway-in-spring-boot-3ea804003021 

Which has a well written explanation of what should be achieved. If you need more context please let me know. 

I'd propose to have some skeleton ready to go and we can connect them up later

File: api-services/auth-service/pom.xml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  <groupId>com.example.authservice</groupId>
+  <artifactId>auth-service</artifactId>
+  <version>0.0.1-SNAPSHOT</version>
+  <parent>
+    <groupId>org.springframework.boot</groupId>
+    <artifactId>spring-boot-starter-parent</artifactId>
+    <version>3.0.0</version>
+    <!-- Or a stable version like 2.6.x if using Java 11 -->
+    <relativePath />
+    <!-- lookup parent from repository -->
```
Comment (M-Byte480) [inline api-services/auth-service/pom.xml:15]: Remove these comments please

File: api-services/post-service/pom.xml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+	<groupId>com.example.postservice</groupId>
+	<artifactId>post-service</artifactId>
+	<version>0.0.1-SNAPSHOT</version>
+	<parent>
+		<groupId>org.springframework.boot</groupId>
+		<artifactId>spring-boot-starter-parent</artifactId>
+		<version>3.0.0</version>
+		<!-- Or a stable version like 2.6.x if using Java 11 -->
+		<relativePath />
+		<!-- lookup parent from repository -->
```
Comment (M-Byte480) [inline api-services/post-service/pom.xml:15]: Same here

**Review by M-Byte480 — CHANGES_REQUESTED** [review]

**Comment by M-Byte480** [comment]
I wish the project had a bit more structure as in the microservices:
image
I am confused on the duplication, and also the apps in the upper half dont seem to read the properties. Is this meant to be a guideline? 

Also I am struggling to see where the redirects would occur. Can you post a screenshot or two on what is supposed to happen?
For me even though you have 8081 specified, it is not read:
image


This is a note for me (and potentially you):
https://medium.com/@ankithahjpgowda/api-gateway-in-spring-boot-3ea804003021

**Comment by M-Byte480** [comment]
I feel like there is a bit of confusion what the API-Gateway is supposed to be doing. 

It takes in the request and maps it to the correct Microservice. It is supposed to be an independent API, not a collection of 3 (which is not your responsibility).

**Comment by M-Byte480** [comment]
Closed as it is no longer used

**Pull request closed without merging**